modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMFieldLocalServiceTest.java

```diff
@@ -323,6 +324,65 @@ public void testSimpleForm() throws Exception {
 		Assert.assertEquals(ddmFormValues, deserializedDDMFormValues);
 	}
 
+	@Test
+	public void testUpdateDDMFormValuesWithLegacyField() throws Exception {
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm("field");
+
+		DDMStructure ddmStructure = _ddmStructureTestHelper.addStructure(
+			ddmForm, StorageType.DEFAULT.toString());
+
+		DDMFormValues ddmFormValues = new DDMFormValues(ddmForm);
+
+		ddmFormValues.setDefaultLocale(LocaleUtil.ENGLISH);
+
+		DDMFormFieldValue ddmFormFieldValue = new DDMFormFieldValue();
+
+		ddmFormFieldValue.setName("field");
+		ddmFormFieldValue.setInstanceId(StringUtil.randomString(8));
+
+		Value value = new LocalizedValue(LocaleUtil.ENGLISH);
+
+		value.addString(
+			LocaleUtil.ENGLISH,
+			LocaleUtil.toLanguageId(LocaleUtil.ENGLISH) + " value");
+
+		ddmFormFieldValue.setValue(value);
+
+		ddmFormValues.addDDMFormFieldValue(ddmFormFieldValue);
+
+		_ddmFieldLocalService.updateDDMFormValues(
+			ddmStructure.getStructureId(), _STORAGE_ID, ddmFormValues);
+
+		List<DDMFormField> ddmFormFields = ddmForm.getDDMFormFields();
```

Comment:
You could assert the `ddmFormValues` before updating the structure, so we know that the value remains the same before and after.